ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/PollableSubscriber.java

```diff
@@ -112,7 +112,7 @@ synchronized boolean isClosed() {
   private void checkRequestTokens() {
     if (tokens == 0) {
       tokens += REQUEST_BATCH_SIZE;
-      makeRequest(REQUEST_BATCH_SIZE);
+      runOnRightContext(() -> makeRequest(REQUEST_BATCH_SIZE));
```

Comment:
fascinating. so without this we always hit the batch limit and then stopped? can we add some tests to make sure we can exceed the `BATCH_SIZE` - this bug is a little embarrassing! 😳